src/main/java/TaskList.java

```diff
@@ -0,0 +1,90 @@
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * TaskList class handles the operations to modify list of tasks given by the user
+ *
+ */
+public class TaskList {
+
+    private ArrayList<Task> taskArrayList;
+
+    /**
+     * Constructor for TaskList with no argument, creates empty ArrayList
+     *
+     */
+    public TaskList() {
+        taskArrayList = new ArrayList<>();
+    }
+
+
+    /**
+     * Adds task to taskArrayList, writes to duke.txt file
+     *
+     * @param storage to write to duke.txt file
+     * @param task to be printed and written
+     * @throws IOException for errors relating to parsing of the file in the storage.writeToFile function
+     */
+    public void addTask(Storage storage, Task task) throws IOException {
+        taskArrayList.add(task);
+        Ui.print("     Got it. I've added this task:\n" + "     " + task.toString() +
+                "\n     Now you have " + taskArrayList.size() + " tasks in the list");
+        storage.writeToFile(task);
```

Comment:
Maybe the Ui.print(...) part can be further abstracted by adding a method for example, Ui.printAddition(task) to enforce a same level of abstraction in the method?